simulator/src/main.rs

```diff
@@ -127,6 +159,70 @@ async fn main() {
                 frame.tick().await;
             }
         }
+        Mode::Json(Json { path, repeat }) => {
```

Comment:
This block is large enough to be it's own function. At which point the other two match arms may as well be split out too.